test/dlc_tests/sanity/quick_checks/test_buildspecs.py

```diff
@@ -0,0 +1,63 @@
+import os
+import re
+
+import pytest
+
+
+@pytest.mark.quick_checks
+@pytest.mark.model("N/A")
+@pytest.mark.integration("buildspecs")
+def test_train_inference_buildspec():
+    """
+    Walk through all buildspecs. Ensure the following:
+    1. "training" or "inference is in the path
+    2. if "training" is in the path, make sure "inference" is not in the file
+    3. if "inference" is in the path, make sure "training" is not in the file
+    """
+    # Look up the path until deep-learning-containers is our base directory
+    dlc_base_dir = os.getcwd()
+    while os.path.basename(dlc_base_dir) != "deep-learning-containers":
+        dlc_base_dir = os.path.split(dlc_base_dir)[0]
+
+    # Regex definitions for buildspec matching
+    buildspec_pattern = re.compile(r"buildspec\S*\.yml")
+    training_pattern = re.compile(r"training", re.IGNORECASE)
+    inference_pattern = re.compile(r"inference", re.IGNORECASE)
+
+    # Navigate through files and look for matches
+    for root, dirnames, filenames in os.walk(dlc_base_dir):
+        for filename in filenames:
+            if buildspec_pattern.search(filename):
+                buildspec_path = os.path.join(dlc_base_dir, root, filename)
+
+                # There are several top level buildspecs, so if we are in the 
+                # top level dir, src dir or eks infra dir, don't count these
+                if (
+                    os.path.split(buildspec_path)[0] != dlc_base_dir
+                    and "eks_infrastructure" not in buildspec_path
```

Comment:
is there no better way to handle this? avoid hardcoding? we need to do handle this everytime a new dir is added

Reply:
Not every time a new dir is added, but every time a buildspec is added without an image type.

We can put allowed dirs in a list if that's preferred.

The other thing we can do is just do a re.match on buildspec instead, but then if a team names a framework buildspec <something>-buildspec.yml, it won't catch it.

Reply:
Resolved by using re.match, and also checking in src/main.py for buildspec format